modules/bpi/bpi.images.inc

```diff
@@ -213,3 +229,71 @@ function bpi_fetch_image($filename, $image_url) {
 
   return file_save_data($result->data, $filename, FILE_EXISTS_REPLACE);
 }
+
+/**
+ * Compute the image type from a field name on a node.
+ *
+ * @param string $image_field_name
+ *   The image field name.
+ * @param array $node
+ *   The node.
+ *
+ * @return NULL|string
+ *   The image type.
+ */
+function bpi_get_image_type($image_field_name, array $node) {
+  /*
+   * We assume that the image type is the last part of the field name, i.e.
+   *   field_{{node type}}_{{image type}}
+   */
+  $pattern = '/field_' . preg_quote($node->type, '/') . '_(?<image_type>.+)/';
```

Comment:
The $node parameter is type hinted to array but accessed as an object. That looks odd to me.